src/Native/configure.cmake

```diff
@@ -175,13 +181,27 @@ check_function_exists(
     kqueue
     HAVE_KQUEUE)
 
-check_function_exists(
-    gethostbyname_r
-    HAVE_GETHOSTBYNAME_R)
+check_prototype_definition(
+     gethostbyaddr_r
+     "int gethostbyaddr_r(const void* addr, const socklen_t len, int type, struct hostent* result, char* buffer, size_t buflen, stuct hostent** entry, int* error)"
+     0
+     "sys/types.h;netdb.h"
```

Comment:
I tested it on OSX and FreeBSD. For some reason clang-3.6 is giving me a compiler switch error on Ubuntu (which is not set in corefx switch by llvm's config file, I am still trying to figure out a way to make it work). I agree that this is something to pay attention to.

From time to time I go through CMakeOutput and CMakeError to ensure that it should fail with expected error. I have also changed the `cmake` command in `src/Native/gen-buildsys-clang.sh` to `cmake debug-trycompile` locally to see what .c[pp] files are produce for introspection in `bin/obj/NetBSD.x64.Debug/Native/CMakeFiles/CMakeTmp/` directory. (without `--debug-trycompile` it deletes those transient generated source, make and object files)